### ISSUE TITLE
File Download: Changing the list of selected files for a multipage file list results in bad download all behavior.

### ISSUE THREAD

**Issue by kcondon** [IQSS/dataverse#3711]
Tested in production with 42 and 108 files, See RT 24178 for details.

If I change the selected file list after downloading then choose download (all) the file downloaded is not a zip file but one of the files on the list. There are at least two ways to reproduce this:

1. Select all without scrolling to see other files. Download. Now scroll to see all files, then select all again so that all are selected. Choose download to get all files. Notice the file downloaded is one of the files, not a zip of all the files.

2. Scroll to see all files, select all, download. Now, deselect one file (2nd from top is what I chose). Now download all and a single file is downloaded, not a zip of all files.

I believe downloading in between changing selected files is essential to reproducing this issue.

**Comment by mheppler** [comment]
Related to **Edit File Metadata: Select All Files in Tab, Can't Edit Them All after clicking Edit Metadata #2859**. Investigate as part of **Spike: Files DataTable, Selection, Paginator vs Scroll #4437**.

**Comment by mheppler** [comment]
Issue #4656, which removed the scrolling feature of the files table, was recently merged as part of [4.9.2](https://github.com/IQSS/dataverse/releases/tag/v4.9.2). I tried to replicate this issue on demo, with a dataset containing 59 files. I was able to select all 59 files, download a ZIP of all the files, then unselect one file, and download a ZIP of the remaining 58 files.

Also, as part of #4656 a new dynamic counter and "Select all ## files in this dataset" text link was added to the top of the files table. This will help you know exactly how files have selected currently. Which comes in very handy as it pertains to the use case reported in this issue.

I suggest that we can close this issue. Will assign to QA and tag @kcondon to confirm and close.
